### PR TITLE
Fix #629 by adding PrivateIPAddress to config.json

### LINKED ISSUE #629: Proposed: save public and private IP of ec2 instances in config.json
to make it possible to use "machine inspect" to get the private IP it would be helpful if it was saved in config.json.  Some systems like consul need to be configured to advertise the private IP and join gossip pools on the private IP space so this would make it easier to script the creation of a consul cluster created with machine.

### PR DESCRIPTION
**Pull request by andrewwatson**
Signed-off-by: Andrew Watson <andrew@ionicsecurity.com>

### CODE REVIEW

File: drivers/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -327,8 +332,8 @@ func (d *Driver) GetIP() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	d.IPAddress = inst.IpAddress
-	return d.IPAddress, nil
+
```
Comment (andrewwatson) [inline drivers/amazonec2/amazonec2.go]: I didn't like that this Getter method was modifying the Driver struct.

Comment (ehazlett) [inline drivers/amazonec2/amazonec2.go]: +1

File: drivers/amazonec2/amazonec2.go [hunk 2]
```diff
@@ -269,12 +270,19 @@ func (d *Driver) Create() error {
...
+	}
 
 	d.waitForInstance()
 
-	log.Debugf("created instance ID %s, IP address %s",
+	log.Debugf("created instance ID %s, IP address %s, Private IP address %s",
 		d.InstanceId,
-		d.IPAddress)
+		d.IPAddress,
+		instance.NetworkInterfaceSet[0].PrivateIpAddress,
```
Comment (nathanleclaire) [inline drivers/amazonec2/amazonec2.go]: Let's use `d.PrivateIPAddress` here instead - it is less verbose and will default to empty string if this field is not present in the response.

File: drivers/amazonec2/amazonec2.go (hunk 1 above)
Comment (nathanleclaire) [inline drivers/amazonec2/amazonec2.go]: +1 if this is called from `docker-machine ip` it does not even save the config so it doesn't matter to set this field in the struct

File: drivers/amazonec2/amazonec2.go (hunk 2 above)
Comment (andrewwatson) [inline drivers/amazonec2/amazonec2.go]: good call.

**Comment by ehazlett** [comment]
LGTM

**Comment by nathanleclaire** [comment]
LGTM

**Pull request merged**